core/src/backend/video/software.rs

```diff
@@ -0,0 +1,73 @@
+//! Pure software video decoding backend.
+
+use crate::backend::render::{BitmapInfo, RenderBackend};
+use crate::backend::video::{
+    EncodedFrame, Error, FrameDependency, VideoBackend, VideoStreamHandle,
+};
+use generational_arena::Arena;
+use swf::{VideoCodec, VideoDeblocking};
+
+/// A single preloaded video stream.
+pub enum VideoStream {}
+
+/// Desktop video backend.
```

Comment:
The word "Desktop" might be confusing here - I guess you meant that it does not utilize the decoders provided by the browser (or anything else), but it _does_ also work on web - it's just that it's pure "internal" code.